shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts

```diff
@@ -647,6 +535,124 @@ export class AppInsightsCore<CfgType extends IConfiguration = IConfiguration> im
                 return _startLogPoller(true);
             };
 
+            function _handleIKeyEndpointPromises(theConfig: IConfiguration) {
+                // app Insights core only handle ikey and endpointurl, aisku will handle cs
+                // But we want to reference these config values so that if any future changes are made
+                // this will trigger the re-run of the watch function
+                // and the ikey and endpointUrl will be set to the new values
+                let ikey = theConfig.instrumentationKey;
+                let endpointUrl = theConfig.endpointUrl; // do not need to validate endpoint url, if it is null, default one will be set by sender
+
+                // Check if we are waiting for previous promises to be resolved, won't apply new changes
+                if (_activeStatus !== eActiveStatus.PENDING) {
```

Comment:
We "probably" should really look at handling reassignment to different promises but that's a much larger (and risker) change.